core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -199,6 +199,11 @@ public Map<String, SnapshotRef> refs() {
     return table().refs();
   }
 
+  @Override
+  public String uuid() {
+    return table().uuid();
```

Comment:
Is this correct? The metadata table is distinct from the underlying table so it seems to me that it shouldn't have the same UUID.

Reply:
I think it is correct, the metadata table is a read only view of certain aspects of a table and ultimately will wrap a `DataTable` .  So I think surfacing the underlying table's UUID is the right thing to do since a metadata table won't ever have a UUID itself right?

Alternatively, we could say since UUIDs aren't even defined for metadata tables and throw an unsupported exception. 
If we do that, we may want to revisit the refs() behavior since that surfaces the underlying table's refs; to be consistent across the APIs we would probably want to throw there as well (although that's a behavior change).

Reply:
I'd probably just generated a new UUID for every metadata table instead of returning the table UUID. That seems more correct to me.

Reply:
@amogh-jahagirdar, any update on this? It should be safer to return a random UUID here.

Reply:
Sorry for the delay, I forgot I had this PR. Just updated!

Reply:
Random UUID is fine but generating random uuid on each call is wrong. For same metadata table, it gives different UUID. 

Fixed in https://github.com/apache/iceberg/pull/9310

Reply:
Yes @ajantha-bhat you're right, there's a consistency issue with this implementation. The same Metadata Table reference should return the same ID.